openedx/core/djangoapps/content/block_structure/tests/helpers.py

```diff
@@ -274,10 +274,14 @@ def create_block_structure(self, children_map, block_structure_cls=BlockStructur
         # create empty block structure
         block_structure = block_structure_cls(root_block_usage_key=self.block_key_factory(0))
 
-        # _add_relation
+        # _add_relation and blocks
         for parent, children in enumerate(children_map):
+            if isinstance(block_structure, BlockStructureBlockData):
+                block_structure._get_or_create_block(self.block_key_factory(parent))  # pylint: disable=protected-access
             for child in children:
                 block_structure._add_relation(self.block_key_factory(parent), self.block_key_factory(child))  # pylint: disable=protected-access
+                if isinstance(block_structure, BlockStructureBlockData):
+                    block_structure._get_or_create_block(self.block_key_factory(child))  # pylint: disable=protected-access
```

Comment:
Why was this necessary (as opposed to invoking block_structure app code to generate this data)?

Reply:
Can you say more, I don't know what you mean by that.  Previously, the test was calling this `create_block_structure` function to create a block structure but not creating the underlying blocks.  This caused an issue when the newly default caching system tried to cache those blocks because they didn't exist even though they were named in the blockstructure object.  I was wondering if there was a different way to generate the whole tree in the first place that would be more "real", is that what you're suggesting?

Reply:
Yeah, I thought that the test would just trigger the normal build process to happen and inspect the results. But if it's not doing that, and you're just continuing the same pattern as existed before, I think that's fine.